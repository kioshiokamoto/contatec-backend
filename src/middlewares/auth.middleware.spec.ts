import { Test, TestingModule } from '@nestjs/testing';
import * as mocks from 'node-mocks-http';
import { AuthMiddleware } from './auth.middleware';

describe('AuthMiddleware', () => {
  let testingModule: TestingModule;
  let service: AuthMiddleware;
  it('Se debe definir el middleware', () => {
    expect(new AuthMiddleware()).toBeDefined();
  });
  beforeEach(async () => {
    const moduleRef = await Test.createTestingModule({
      providers: [AuthMiddleware],
    }).compile();
    service = moduleRef.get(AuthMiddleware);
  });
  it('Debe matchear con snapshot', () => {
    const req = mocks.createRequest();
    const res = mocks.createResponse();
    const demo = service.use(req, res, () => {
      return;
    });
    expect(service.use).toMatchSnapshot();
  });
});
