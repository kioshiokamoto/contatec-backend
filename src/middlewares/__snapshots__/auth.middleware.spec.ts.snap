// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthMiddleware Debe matchear con snapshot 1`] = `[Function]`;
